charts/kyverno/templates/deployment.yaml

```diff
@@ -69,7 +69,7 @@ spec:
         {{- toYaml .Values.extraInitContainers | nindent 8 }}
       {{- end }}
         - name: kyverno-pre
-          image: {{ .Values.initImage.repository }}:{{ default .Chart.AppVersion (default .Values.image.tag .Values.initImage.tag) }}
+          image: {{ include "kyverno.image" (dict "image" .Values.initImage "defaultTag" .Chart.AppVersion) | quote }}
```

Comment:
The `initImage.tag` appears to no longer gets used which is not great.  I think the init image `defaultTag` value might need to still have something like `default .Chart.AppVersion .Values.initImage.tag` and then in the helper template it will still default to `image.tag`.  Maybe change to this:

```
{{ include "kyverno.image" (dict "image" .Values.initImage "defaultTag" (default .Chart.AppVersion .Values.initImage.tag)) | quote }}
```

Reply:
It is used within the template: 

`{{ .image.registry }}/{{ required "An image repository is required" .image.repository }}:{{ default .defaultTag .image.tag }}`

What I think is missing is that the initImage should use the `image.tag` if the `initImage.tag` is not available. As described in helm `values.yaml` :

```yaml
# -- Image tag
# If initImage.tag is missing, defaults to image.tag
tag:  # replaced in e2e tests
```

This will translate into a very small change on your suggestion:
`{{ include "kyverno.image" (dict "image" .Values.initImage "defaultTag" (default .Chart.AppVersion .Values.image.tag)) | quote }}
`

What do you think ? 🙂

Reply:
```sh
kyverno/charts/kyverno st-add-registry-on-helm-chart*​​​ 
❯ helm template kyverno . --set image.registry="custom.registry" --set image.repository="kyverno" --set initImage.registry="custom.registry" --set initImage.repository="kyvernopre" --set image.tag="1.7.4" | grep -C1  custom.registry
        - name: kyverno-pre
          image: "custom.registry/kyvernopre:1.7.4"
          imagePullPolicy: IfNotPresent
--
        - name: kyverno
          image: "custom.registry/kyverno:1.7.4"
          imagePullPolicy: IfNotPresent

kyverno/charts/kyverno st-add-registry-on-helm-chart*​​​ 
❯ helm template kyverno . --set image.registry="custom.registry" --set image.repository="kyverno" --set initImage.registry="custom.registry" --set initImage.repository="kyvernopre" --set image.tag="1.7.4" --set initImage.tag="1.7.6" | grep -C1  custom.registry
        - name: kyverno-pre
          image: "custom.registry/kyvernopre:1.7.6"
          imagePullPolicy: IfNotPresent
--
        - name: kyverno
          image: "custom.registry/kyverno:1.7.4"
          imagePullPolicy: IfNotPresent
```